implementations/elixir/ockam/ockam/lib/ockam/transport/portal/inlet_worker.ex

```diff
@@ -97,4 +96,10 @@ defmodule Ockam.Transport.Portal.InletWorker do
     :ok = :gen_tcp.close(socket)
     {:stop, :normal, state}
   end
+
+  defp send_over_tcp(data, %{socket: socket, tcp_wrapper: tcp_wrapper}) do
+    tcp_wrapper.wrap_tcp_call(fn ->
+      :gen_tcp.send(socket, data)
+    end)
```

Comment:
What do you think about receiving this argument in as MFA syntax, as in:
 ```suggestion
    tcp_wrapper.wrap_tcp_call([:gen_tcp, :send, [socket, data])
```
similar to what you'd use with `Kernel.apply/3`? I think this change might skip us from allocating an anonymous-function closure with ever-changing captures, that would need to get garbage collected later.

This has some other tradeoffs as well:
- Either preventing use of an anonymous function, or requiring two function heads in `Ockam.Transport.TCP.DefaultWrapper` to permit both styles
- In cases where consumers of the library use hot-code-reloading, anonymous functions will use whatever was in place when the call-site occurred, but apply-calls would always use the then-latest definition of that module + function when the execution actually happens, which is kind of a weird nuance sometimes.

Reply:
Yeah, that's definitely necessary. I changed it to use `Kernel.apply/3`